cabal-testsuite/PackageTests/BuildToolDependsNonComponent/setup.test.hs

```diff
@@ -0,0 +1,4 @@
+import Test.Cabal.Prelude
+-- Test build-tool-depends between two packages
+main = cabalTest $ do
+    cabal "v2-build" ["client", "--disable-per-component"]
```

Comment:
Even the test fails now, it will still fail as this flag says that `bad-do-not-build-me` have to be built too.

Will this test still fail as expected, if `bad-do-not-build-me` executable is removed?

Reply:
Yes. Removing the `bad-do-not-build-me` executable results in the following:

```
% cabal new-clean && cabal new-build client --disable-per-component
Resolving dependencies...
Build profile: -w ghc-8.6.5 -O1
In order, the following will be built (use -v for more details):
 - pre-proc-999.999.999 (exe:pre-proc) (first run)
 - client-0.1.0.0 (exe:hello-world) (first run)
Configuring pre-proc-999.999.999...
cabal: Unknown build target 'exe:pre-proc'.
There is no executable component 'pre-proc'.

cabal: Failed to build pre-proc-999.999.999 (which is required by
client-0.1.0.0).
```

I will make the change.

Reply:
Ok, now I see why there was second executable. We should not build it. Argh, this is tricky. I'll take this over to see how to solve this without compromises.